deployment/ccip/shared/opsutil/evm.go

```diff
@@ -0,0 +1,146 @@
+package opsutil
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/deployergroup"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+
+	mcmslib "github.com/smartcontractkit/mcms"
+	mcmssdk "github.com/smartcontractkit/mcms/sdk"
+	mcmstypes "github.com/smartcontractkit/mcms/types"
+)
+
+// Addressable is an interface that provides a method to get the address of a contract.
+type Addressable interface {
+	Address() common.Address
+}
+
+// EVMCallDeps contains the dependencies required for an EVM call operation.
+// It includes the contract bindings, the chain on which the operation is performed, and transaction options.
+type EVMCallDeps[C Addressable] struct {
+	Contract C
```

Comment:
Any benefit in generics for Contract? since its EVM specific couldnt we just do contract binding?
It'd slightly improve the readability